packages/extension-ui/src/components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019-2020 @polkadot/extension-ui authors & contributor
+// SPDX-License-Identifier: Apache-2.0
+
+import { WithTranslation } from 'react-i18next';
+
+import React from 'react';
+// import styled from 'styled-components';
+
+import translate from './translate';
+import Header from '../partials/Header';
+import ActionText from './ActionText';
+// import { ThemeProps } from '../types';
+
+interface Props extends WithTranslation {
+  children: React.ReactNode;
+  className?: string;
+  error?: Error | null;
+  onError?: () => void;
+  trigger?: unknown;
+}
+
+interface State {
+  error: Error | null;
+  prevTrigger: string | null;
+}
+
+// NOTE: This is the only way to do an error boundary, via extend
+class ErrorBoundary extends React.Component<Props> {
+  state: State = { error: null, prevTrigger: null };
+
+  static getDerivedStateFromError (error: Error): Partial<State> {
+    return { error };
+  }
+
+  static getDerivedStateFromProps ({ trigger }: Props, { prevTrigger }: State): State | null {
+    const newTrigger = JSON.stringify({ trigger });
+
+    return (prevTrigger !== newTrigger)
+      ? { error: null, prevTrigger: newTrigger }
+      : null;
+  }
+
+  public componentDidCatch (): void {
+    const { onError } = this.props;
+
+    onError && onError();
+  }
+
+  public render (): React.ReactNode {
+    const { children, error: errorProps, t } = this.props;
+    const { error } = this.state;
+    const displayError = errorProps || error;
```

Comment:
This is the only place where it _may_ be an issue, but since we don't pass props through... also stumped just looking at the code.

Reply:
Since the state is derived from the props, the way I understand it is that `displayError === error`. I removed it now and just use the error from the state, but I still need to click cancel twice to be redirected.